ci-operator/step-registry/fips-check/optional-operator-images-scan/fips-check-optional-operator-images-scan-commands.sh

```diff
@@ -0,0 +1,194 @@
+for package in ${package_list}; do
+    echo "Starting scan for ${package}"
+    mkdir "${data_dir}/${package}" || true
+    skippedCSVs="compliance-operator.v0.1.32|compliance-operator.v0.1.61|file-integrity-operator.v0.1.13|file-integrity-operator.v0.1.32|file-integrity-operator.v1.0.0"
+    currentCSVs=$(cat /tmp/$catalog.json | jq '.items[] | select(.metadata.name=="'${package}'").status.channels[].currentCSV'| uniq | grep -Ev $skippedCSVs || true);
+    currentCSVs=${currentCSVs//\"/}
+    for csv in ${currentCSVs}; do
+        images_test_result="${data_dir}/${package}/$csv"
+        echo "##Starting scan for csv ${csv}" >> $images_test_result
+        image_list=$(cat /tmp/$catalog.json | jq '.items[].status.channels[]|select(.currentCSV=="'$csv'")|.currentCSVDesc.relatedImages[]' || true);
+	    image_list=${image_list//,/}
+	    image_list=${image_list//\"/}
+        if $internal; then
+            image_list=$(echo $image_list | sed -r 's/registry.redhat.io/brew.registry.redhat.io/g' )
+            image_list=$(echo $image_list | sed -r 's/registry.stage.redhat.io/brew.registry.stage.redhat.io/g')
+        fi
+        for image_url in ${image_list}; do
+            imges_test_result=$(ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
+                "sudo podman run --privileged -it -v /tmp/tmp/:/root/:ro registry.ci.openshift.org/ci/check-payload scan operator --spec $image_url" || true)
+            echo "###test result for $image_url is: $imges_test_result" >> $images_test_result
+        done
+        echo "cat ${imges_test_result}" >> ${scan_result_file}
+        resFail=$(grep -E "Failure Report" ${images_test_result} || true)
+        resSuessWithWarnings=$(grep -E "Successful run with warnings" ${images_test_result} || true)
+        if [[ -n $resFail ]];then
+            echo "Fips check  for csv $csv in package $package failed!"
+            echo "$csv" >> ${scan_result_failure_csvs}
+            pass=false
+        elif [[ -n $resSuessWithWarnings ]];then
+            echo "Fips check  for csv $csv in package $package succeed with warnings!"
+            echo "$csv" >> ${scan_result_succeed_with_warnings_csvs}
+            pass=false
+        else
+            echo "Fips check for csv $csv in package $package succeed!"
+        fi
+        #delete containers and images to save disk space
+        echo "delete containers"
+        ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
+            "for container in \$(sudo podman ps -q -a); do sudo podman rm \${container} || true; done"
+        echo "delete images"
+        ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
+            "for image_url in \${image_list}; do sudo podman rmi \${image_url} || true; done"
```

Comment:
Instead of delete the image one by one, you can delete them in batch `podman rmi image1 image2 imageN`

Reply:
Updated